pkg/app/mutate_app.go

```diff
@@ -142,6 +150,12 @@ func (m *Mutator) mutateKubeConfig(ctx context.Context, app v1alpha1.App) ([]mut
 		return nil, nil
 	}
 
+	// Return early if kubeconfig not found.
+	_, err := m.k8sClient.K8sClient().CoreV1().Secrets(app.Namespace).Get(ctx, key.ClusterKubeConfigSecretName(app), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return nil, nil
```

Comment:
If admission-controller returns early at this point, When it would set the default value for these app CRs? Perhaps in the next reconciliation?

Reply:
The mutation webhook fires before validation. So creating the app CR will be blocked until the next reconciliation. Once the kubeconfig exists the app CR can be created.

I've tested this with cluster-operator and it works but I still need to check that no cluster-operator alerts fire.

Reply:
> but I still need to check that no cluster-operator alerts fire. 

Sure, I think it's better to check with this first.

Reply:
Cool will test that with cluster-operator now.

Reply:
Yes the initial app creation fails because the kubeconfig does not exist. It works when it is retried later so cluster creation is fine but the operatorkit error rate alerts will fire.

My idea is to handle the validation errors in cluster-operator and here to check the app CR version and skip defaulting and validation for older versions.

This way we can link the rollout of the admission controller to a platform release. WDYT?



```
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.NewUpdatePatch computing App CRs to create  | resource/v2/appresource/create.go:50 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.NewUpdatePatch computed 11 App CRs to create | resource/v2/appresource/create.go:58 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.NewUpdatePatch computing App CRs to delete | resource/v2/appresource/delete.go:49 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.NewUpdatePatch computed 0 App CRs to delete | resource/v2/appresource/delete.go:57 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.NewUpdatePatch computing App CRs to update | resource/v2/appresource/update.go:46 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.NewUpdatePatch computed 0 App CRs to update | resource/v2/appresource/update.go:57 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
D 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.ApplyCreateChange creating App CR `cert-exporter` in namespace `8g0ve` | resource/v2/appresource/create.go:22 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
W 11/13 10:34:57 /apis/cluster.x-k8s.io/v1alpha2/namespaces/default/clusters/8g0ve app.ApplyCreateChange retrying due to error | operatorkit/v2/pkg/resource/wrapper/retryresource/crud_resource.go:161 | controller=cluster-operator-cluster-controller | event=update | loop=2440 | version=318909568
	/go/pkg/mod/github.com/giantswarm/operatorkit/v2@v2.0.1/pkg/resource/wrapper/retryresource/crud_resource.go:154
	/go/pkg/mod/github.com/giantswarm/resource/v2@v2.0.0/appresource/create.go:28
	unknown: admission webhook "apps.app-admission-controller-unique.giantswarm.io" denied the request: validation error: kubeconfig secret `8g0ve-kubeconfig` in namespace `8g0ve` not found
```

Reply:
It makes sense, good to point out that we need to see the version of cluster-operator.

Reply:
Error matching is working now. So alerts will not fire in cluster-operator.